src/app/scenario/scenario.component.ts

```diff
@@ -339,31 +339,24 @@ export class ScenarioComponent implements OnInit {
 
   cancelEdit() {
     this.reloadSteps();
+    this.editingStep.content = this.AddedStepsContents[this.editingIndex];
+    this.editingStep.title = this.AddedStepsTitle[this.editingIndex];
     this.editOpen = false;
     this.editModal.close();
   }
 
   reloadSteps() {
-    if (this.contentCancelStep == '')
-      this.scenarioService.get(this.selectedscenario.id).subscribe(
-        (scenario) => {
-          this.selectedscenario.steps = scenario.steps;
-        },
-        (e: HttpErrorResponse) => {
-          this.alertDanger('Error deleting object: ' + e.error.message);
-          this.editingStep.content = this.contentCancelStep;
-          this.editingStep.title = this.titleCancelStep;
-          while (this.stepsToBeAdded > 0) {
-            this.stepsToBeAdded--;
-            this.selectedscenario.steps.pop();
-          }
-        }
-      );
-    else {
-      this.editingStep.content = this.contentCancelStep;
-      this.editingStep.title = this.titleCancelStep;
+    while (this.stepsToBeAdded > 0) {
+      this.stepsToBeAdded--;
+      this.selectedscenario.steps.pop();
     }
   }
+  scenarioHasVM(scenario: Scenario): boolean {
+    if (this.selectedscenario.virtualmachines.length != 0)
+      if (Object.keys(this.selectedscenario.virtualmachines[0]).length != 0)
```

Comment:
Do you maybe want to check all VMSets, not only the first one? There should be no emty VMSet, all VMSets should have at least one VM and we should have at least one VMSet

Reply:
I will see what I can do.

Reply:
Done.

Reply:
You might want to check this again.
What happens if the last VMSet is empty?

So create one VMSet, add one VM, cerate another one and let it be empty

Reply:
Done, I hope this solution is better